revoke/revoke.go

```diff
@@ -123,69 +213,141 @@ func getIssuer(cert *x509.Certificate) *x509.Certificate {
 	}
 
 	return issuer
-
 }
 
-// check a cert against a specific CRL. Returns the same bool pair
-// as revCheck.
-func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
-	crl, ok := CRLSet[url]
+// checks whether CRL in memory is valid
+func (r *Revoke) isInMemoryCRLValid(key string) bool {
+	defer r.lock.Unlock()
+	r.lock.Lock()
+	crl, ok := r.crlSet[key]
 	if ok && crl == nil {
 		ok = false
-		delete(CRLSet, url)
+		delete(r.crlSet, key)
 	}
 
-	var shouldFetchCRL = true
-	if ok {
-		if !crl.HasExpired(time.Now()) {
-			shouldFetchCRL = false
+	if ok && !crl.HasExpired(time.Now()) {
```

Comment:
I think this would be better:

```go
        if ok && crl == nil {
                ok = false
                delete(r.crlSet, key)
        } else if crl == nil {
                delete(r.crlSet, key)
                return false
        }

        if ok && !crl.HasExpired(time.Now()) {
                return true
        }
```